src/mongocxx/test/collection.cpp

```diff
@@ -2665,4 +2667,56 @@ TEST_CASE("bulk_write with container", "[collection]") {
     REQUIRE(result->inserted_count() == 10);
     REQUIRE(collection.count_documents({}) == 10);
 }
+
+/* Regression test for CXX-2028. */
+TEST_CASE("find_and_x operations append write concern correctly", "[collection]") {
+    instance::current();
+    mongocxx::client client{uri{}};
+    mongocxx::write_concern wc;
+    wc.acknowledge_level(mongocxx::write_concern::level::k_acknowledged);
+
+    mongocxx::collection collection = client["fam_wc"]["collection"];
+    collection.drop();
```

Comment:
Just FYI, there's an uncaught `mongocxx::operation_exception` here and in a lot of the methods below. I'm not exactly sure if you'd want to handle that or leave it as is. Most of the code looks like it ignores exceptions with a few commands passed to `REQUIRE_NOTHROW`

Reply:
Oh good point. I thought that might be a bug, since we should be ignoring "ns not found" errors. But the driver will not throw for "ns not found": https://github.com/mongodb/mongo-cxx-driver/blob/060b6dd5fc23eade27710a30e6a12eeecb84e38a/src/mongocxx/collection.cpp#L1196-L1203

If an exception was thrown, that would be unexpected and should fail the test. So I think this is ok.